src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -427,17 +427,14 @@ private void ReadToken(string encodedJson)
         {
             // JWT must have 2 dots
             Dot1 = encodedJson.IndexOf('.');
-            if (Dot1 == -1)
-                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14100, encodedJson)));
-
-            if (Dot1 == encodedJson.Length)
-                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14100, encodedJson)));
+            if (Dot1 == -1 || Dot1 == encodedJson.Length - 1)
+                throw LogHelper.LogExceptionMessage(new SecurityTokenMalformedTokenException(LogHelper.FormatInvariant(LogMessages.IDX14100, encodedJson)));
 
             Dot2 = encodedJson.IndexOf('.', Dot1 + 1);
             if (Dot2 == -1)
-                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14120, encodedJson)));
+                throw LogHelper.LogExceptionMessage(new SecurityTokenMalformedTokenException(LogHelper.FormatInvariant(LogMessages.IDX14120, encodedJson)));
 
-            if (Dot2 == encodedJson.Length)
+            if (Dot2 == encodedJson.Length - 1)
```

Comment:
>if (Dot2 == encodedJson.Length - 1) [](http://example.com/codeflow?start=12&length=35)

This change is unrelated with the exceptions thrown, are we fixing a bug here?

Reply:
Yes, this is a bug that should be fixed.

Reply:
Sounds good, lets make sure the commit message also reflects we are doing this fix.

Reply:
I already committed and updated the branch. I think I can update the comment when it is ready to be merged.